crates/llama-cpp-bindings/build.rs

```diff
@@ -26,7 +29,14 @@ fn main() {
     if cfg!(feature = "cuda") {
         config.define("LLAMA_CUBLAS", "ON");
         config.define("CMAKE_POSITION_INDEPENDENT_CODE", "ON");
-        println!("cargo:rustc-link-search=native=/usr/local/cuda/lib64");
+        if cfg!(target_os = "windows") {
+            let Ok(cuda_path) = env::var("CUDA_PATH") else {
+                panic!("CUDA_PATH is not set");
+            };
+            println!(r"cargo:rustc-link-search=native={}\lib\x64", cuda_path);
+        } else {
+            println!("cargo:rustc-link-search=native=/usr/local/cuda/lib64");
+        }
         println!("cargo:rustc-link-lib=cudart");
         println!("cargo:rustc-link-lib=culibos");
```

Comment:
I did some search work, and compared the cuda doc carefully and found:

Until `12.0.0`, there indeed exists `culibos` regardless of os type, it is a thread abstraction layer, as mentioned here: https://docs.nvidia.com/cuda/archive/12.0.0/cusparse/index.html#static-library-support

Since `12.0.1`, in the similar page, there's no such keyword found: https://docs.nvidia.com/cuda/archive/12.0.1/cusparse/index.html#static-library-support

I do suspect this lib has been removed from `12.0.1` and afterwards, that's why I failed to compile on my laptop cause I installed latest cuda tookit (`12.3.1`).

So this line should continue to work on release build until we update settings.

As for local development, we can raise a separate PR to fix it for both windows & *nix platforms.

Reply:
as long as the ci build pass - we're good

Whether to fix the build for cuda12 can be discussed separately